app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb

```diff
@@ -30,6 +30,10 @@ def parse
     sshkeys
   end
 
+  def architecture(image_id)
+    img_to_arch[image_id] ||= collector.image(image_id).specifications.architecture
```

Comment:
@jaywcarman I'm assuming the issue that some instances are based on images which no longer are returned by `GET /images` is that right?

Since multiple instances could be provisioned from the same image this could result in the same API call being issues for the same image multiple times.

I never liked the `img_to_arch` hash, I think we should encapsulate this in the collector like:
```
def image(image_id)
  return images[image_id] if images[image_id] # NOTE this assumes each item returned by `GET /images` is equal to the item returned by `GET /images/:id`

  images[image_id] = images_api.pcloud_cloudinstances_images_get(cloud_instance_id, image_id)
end

def image_architecture(image_id)
  images[image_id]&.specifications&.architecture
end
```

If `GET /images` isn't just an array of what `GET /images/:id` would return then we'll have to keep the `@images` and whatever each `image()` call would return separate.

Reply:
> NOTE this assumes each item returned by `GET /images` is equal to the item returned by `GET /images/:id`

@agrare The image objects returned by `GET /images` ([pcloud-cloudinstances-images-getall](https://cloud.ibm.com/apidocs/power-cloud#pcloud-cloudinstances-images-getall)) contain a subset of the attributes returned by `GET /images/:id` ([pcloud-cloudinstances-images-get](https://cloud.ibm.com/apidocs/power-cloud#pcloud-cloudinstances-images-get)), _so_ I think we're safe to go with your idea to put all of the logic directly in the collector and minimize API calls. That seems to be an all-around better approach, so I'll push another patch up soon.

That said, in the existing patch I thought since we memoized the `img_to_arch` hash [here](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb#L18) that the `collector.image(image_id).specifications.architecture` would only occur the first time that `img_to_arch[image_id]` was `nil`? Was I mistaken?